### PR TITLE
boolean xnor implementation ts and python impl

### PR DESCRIPTION
**Pull request by gaudyb**

### CODE REVIEW

File: javascript/core/src/verbs/util/boolean-logic.ts [hunk 1]
```diff
@@ -87,25 +89,28 @@ export function and(comparisons: (1 | 0 | null)[]): 1 | 0 | null {
  * @returns
  */
 export function xor(comparisons: (1 | 0 | null)[]): 1 | 0 | null {
```
Comment (natoverse) [inline javascript/core/src/verbs/util/boolean-logic.ts:91]: Should we replace the bulk of this function with the even/odd logic to simplify?

File: javascript/core/src/verbs/util/boolean-logic.ts [hunk 2]
```diff
@@ -156,3 +161,20 @@ export function nand(comparisons: (1 | 0 | null)[]): 1 | 0 | null {
 	}
 	return null
 }
+
+/**
+ * Logical XNOR
+ * @param comparisons
+ * @returns
+ */
+ export function xnor(comparisons: (1 | 0 | null)[]): 1 | 0 | null {
```
Comment (natoverse) [inline javascript/core/src/verbs/util/boolean-logic.ts:170]: Please double-check that the code comments on each of these functions matches with our latest and most accurate descriptions of the logic.

**Review by natoverse — APPROVED** [review]
I added a couple of comments that I think would be work addressing, but fine to merge once you get there

File: javascript/core/src/verbs/util/boolean-logic.ts [hunk 3]
```diff
@@ -87,25 +89,28 @@ export function and(comparisons: (1 | 0 | null)[]): 1 | 0 | null {
...
-			// shortcut more than one true immediately
-			return 0
-		}
-		if (comparisons[i] === null) {
-			nulls++
-		}
+	let xor = null
+
+	if(comparisons.length > 0){
+		if(comparisons[0] === null)
```
[the remark this answers is not in the document]
Comment (gaudyb) [reply javascript/core/src/verbs/util/boolean-logic.ts:95]: it's not redundant because the loop starts at 1

**Pull request merged**